instance/models/mixins/ansible.py

```diff
@@ -20,98 +20,63 @@
 Instance app model mixins - Ansible
 """
 import os
+from collections import namedtuple
 from django.conf import settings
-from django.core.validators import MinValueValidator
 from django.db import models
-from django.template import loader
 
 from instance import ansible
 from instance.repo import open_repository
 from instance.utils import poll_streams
 
 
-class AnsibleInstanceMixin(models.Model):
+Playbook = namedtuple('Playbook', [
+    'source_repo',  # Path or URL to a git repository contianing the playbook to run
+    'playbook_filename',  # Relative path to the playbook within source_repo
+    'requirements_path',  # Relative path to a python requirements file to install before running the playbook
+    'version',  # the git tag/commit hash/branch to use
+    'variables',  # a YAML string containing extra variables to pass to ansible when running this playbook
+])
+
+
+class AnsibleAppServerMixin(models.Model):
```

Comment:
@bradenmacdonald Currently, yes, we use one playbook per type of VM, so that could work for now. But I'm pointing out that conceptually this doesn't match the scope of playbooks in ansible, and there might be situations where this assumption will come back to bite us.